### PR TITLE
Cu 225rxvv in game debug mode

### PR DESCRIPTION
**Pull request by Izenz**
Debug Mode window for PLAY builds.
Provisional version for VS1.

### CODE REVIEW

**Comment by Bernatmago** [comment]
Task linked: [CU-225rxvv In-game Debug mode](https://app.clickup.com/t/225rxvv)

File: Source/src/modules/ModuleDebugMode.cpp [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+			glPolygonMode(GL_FRONT_AND_BACK, GL_LINE);
+		}
+		ImGui::Separator();
+
+		ImGui::Text("Player position: ");
+		if (ImGui::DragFloat3("(x, y, z)", player_pos_editor.ptr(), 0.1f, -inf, inf))
+		{
+			player->GetTransform()->SetGlobalPosition(player_pos_editor);
+		}
+	}
```
Comment (VBoschMatas) [inline Source/src/modules/ModuleDebugMode.cpp:154]: We could add in a next PR buttons to enable-disable debug draw and some other options from the config window, but 100% optional right now.

**Review by VBoschMatas — APPROVED** [review]
Looks nice, good job

File: Source/src/Application.cpp [hunk 2]
```diff
@@ -34,7 +34,8 @@ Hachiko::Application::Application()
     modules.push_back(debug_draw = new ModuleDebugDraw());
     modules.push_back(editor = new ModuleEditor());
     modules.push_back(event = new ModuleEvent());
-    modules.push_back(ui = new ModuleUserInterface());    
+    modules.push_back(ui = new ModuleUserInterface()); 
+	modules.push_back(debug_mode = new ModuleDebugMode());
```
Comment (Bernatmago) [inline Source/src/Application.cpp:38]:
```suggestion
modules.push_back(debug_mode = new ModuleDebugMode());
```

File: Source/src/utils/Timer.cpp [hunk 3]
```diff
@@ -149,5 +155,17 @@ double Hachiko::EngineTimer::Update()
     prev_tick_time = tick_time;
     if (running)
         total_time = timer.Read();
+    CountFPS();
     return delta_time;
 }
+
+void Hachiko::EngineTimer::CountFPS()
```
Comment (Bernatmago) [inline Source/src/utils/Timer.cpp:162]: This might not be needed

**Review by Bernatmago — APPROVED** [review]

File: Source/src/utils/Timer.cpp (hunk 3 above)
Comment (Izenz) [reply Source/src/utils/Timer.cpp:162]: Indeed, it is not.

**Pull request merged**